data/restapi/src/main/java/tech/pegasys/artemis/restapi/RestApi.java

```diff
@@ -0,0 +1,125 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.restapi;
+
+import static tech.pegasys.artemis.provider.JsonProvider.objectToJSON;
+
+import com.google.common.primitives.UnsignedLong;
+import io.javalin.Javalin;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.datastructures.state.BeaconState;
+import tech.pegasys.artemis.networking.p2p.JvmLibP2PNetwork;
+import tech.pegasys.artemis.networking.p2p.api.P2PNetwork;
+import tech.pegasys.artemis.storage.ChainStorageClient;
+
+public class RestApi {
+
+  private Javalin app;
+  private ChainStorageClient chainStorageClient;
+  private JvmLibP2PNetwork network;
+  boolean isLibP2P = false;
+
+  public RestApi(
+      ChainStorageClient chainStorageClient, P2PNetwork p2pNetwork, final int portNumber) {
+    this.chainStorageClient = chainStorageClient;
+    if (p2pNetwork instanceof JvmLibP2PNetwork) {
+      isLibP2P = true;
+      network = (JvmLibP2PNetwork) p2pNetwork;
+    }
+    app = Javalin.create().start(portNumber);
+
+    setGenesisTimeHandler();
+    setBeaconHeadHandler();
+    setBeaconBlockHandler();
+    setBeaconStateHandler();
+    setPeerIDHandler();
+    setPeersHandler();
+    setENRHandler();
+  }
+
+  private void setGenesisTimeHandler() {
+    Map<String, Long> jsonObject = new HashMap<>();
+    jsonObject.put("genesis_time", chainStorageClient.getGenesisTime().longValue());
+    app.get("/node/genesis_time", ctx -> ctx.json(jsonObject));
+  }
+
+  private void setBeaconHeadHandler() {
+    app.get(
+        "/beacon/head",
+        ctx -> {
+          Bytes32 head_block_root = chainStorageClient.getBestBlockRoot();
+          Bytes32 head_state_root = chainStorageClient.getBestBlockRootState().hash_tree_root();
+          UnsignedLong head_block_slot = chainStorageClient.getBestSlot();
+          Map<String, Object> jsonObject = new HashMap<>();
+          jsonObject.put("slot", head_block_slot.longValue());
+          jsonObject.put("block_root", head_block_root.toHexString());
+          jsonObject.put("state_root", head_state_root.toHexString());
+          ctx.json(jsonObject);
+        });
+  }
+
+  // Skipping implementing block querying by slot for now
+  private void setBeaconBlockHandler() {
+    app.get(
+        "/beacon/block",
+        ctx -> {
+          Bytes32 root = Bytes32.fromHexString(ctx.queryParam("root"));
+          BeaconBlock block = chainStorageClient.getStore().getBlock(root);
+          ctx.result(objectToJSON(block)).contentType("application/json");
+        });
+  }
+
+  // Skipping implementing state querying by slot for now
+  // NOTE: here the root is beacon block root, in parallel to how we index beacon state's in Store
```

Comment:
fyi, querying by slot is more useful (from WB testing perspective) than querying by block hash.  so it would def be good to include next revision

Reply:
Yeah, the only reason I didn't do that is, its a bit expensive because you can't find the specific block/state for that slot, you need to get the canonical head and then get its ancestor at that slot. I'll get it done Monday morning tho.

Reply:
Ya, I figured.  Just giving u some context on how it will be used in the near future

Reply:
We'll need an efficient lookup by slot to respond to Req/Resp requests as well.  So would be good to maintain the canonical mapping rather than having to walk back from HEAD all the time (walking backwards would also expose a DOS vector by requesting very old slots).

Reply:
I'm not sure if there is a canonical mapping. Depending on your current head, the block for that slot might have been empty or not, and in an even more extreme case may be a totally different block.

Reply:
I probably should have said the "current canonical mapping".  Basically maintaining a mapping from slot number to the block from the current best chain at that slot (if there is one).  When fork choice results in selecting a different chain as the best we'd have to walk back to the point where the current and new chains diverged updating the mapping.

That's a routine mapping for ETH1 clients to maintain so they can support things like `getBlockByNumber` so if we're going to be able to get by slot I suspect we'll need a similar mapping.

Reply:
That makes sense.